File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/IntegerEnumAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;
import java.util.List;

/**
 * Represents a descriptor for an attribute with an integer enum value.
 */
public class IntegerEnumAttributeDescriptor extends AttributeDescriptor {
  private List<Long> enumeration;

  /**
   * Constructs an IntegerEnumAttributeDescriptor object with the specified parameters.
   * @param displayName
   * @param technicalName
   * @param description
   * @param required
   * @param enumeration
   */
  public IntegerEnumAttributeDescriptor(
      String displayName,
      String technicalName,
      String description,
      Boolean required,
      List<Long> enumeration) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public static IntegerEnumAttributeDescriptor fromProto(
      IdentityEntitiesV2.IntegerEnumAttributeDefinitionV2 res) {
    return new IntegerEnumAttributeDescriptor(
        res.getDisplayName(),
        res.getId(),
        res.getDescription(),
        res.getRequired(),
        res.getEnumList());
  }

  public IdentityEntitiesV2.IntegerEnumAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.IntegerEnumAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .addAllEnum(this.enumeration)
        .build();
  }
}

File: languages/js/src/entity/identity_v2/integer_enum_attribute_descriptor.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { AttributeDescriptor } from "./attribute_descriptor";

/**
 * Represents a descriptor for an attribute with an integer enum value.
 */
export class IntegerEnumAttributeDescriptor extends AttributeDescriptor {
  enumeration: number[];

  /**
   * Constructs an IntegerEnumAttributeDescriptor object with the specified parameters.
   * @param displayName 
   * @param technicalName 
   * @param description 
   * @param required 
   * @param enumeration 
   */
  constructor(
    displayName: string,
    technicalName: string,
    description: string,
    required: boolean,
    enumeration: number[]
  ) {
    super(displayName, technicalName, description, required);
    this.enumeration = enumeration;
  }

  public toProto(): identityEntitiesProto.IntegerEnumAttributeDefinitionV2 {
    return identityEntitiesProto.IntegerEnumAttributeDefinitionV2.fromPartial({
      displayName: this.displayName,
      id: this.technicalName,
      description: this.description,
      required: this.required,
      enum: this.enumeration
    });
  }

  static fromProto(
    r: identityEntitiesProto.IntegerEnumAttributeDefinitionV2
  ): IntegerEnumAttributeDescriptor {
    return new IntegerEnumAttributeDescriptor(
      r.displayName,
      r.id,
      r.description,
      r.required,
      r.enum
    );
  }
}
